src/pages/UserSettings/UserSettings.tsx

```diff
@@ -0,0 +1,209 @@
+import Row from "../../components/layout/Row";
+import ConfirmationModal from "../../components/modals/ConfirmationModal";
+import { Constants } from "../../constants";
+import { useApi } from "../../contexts/ApiProvider";
+import { useUser } from "../../contexts/UserProvider";
+import { SettingShort } from "../../data/models/Setting";
+export default function UserSettings() {
+  const [toggled, setToggled] = useState(false);
+  const { user, deleteUser } = useUser();
+  const { api } = useApi();
+  const [hasEmailAlerts, setHasEmailAlerts] = useState(
+    user!.settings.find((s) => s.key === "notify" && s.enabled === true) !==
+      undefined,
+  );
+  // we only care about the time of this date
+  const [emailTime, setEmailTime] = useState(
+    user!.settings.find((s) => s.key === "notify")?.value ?? "09:00",
+  );
+  const [email, setEmail] = useState("");
+  const [changeEmailState, setChangeEmailState] = useState("");
+
+  const [username, setUsername] = useState("");
+  const [changeUsernameState, setChangeUsernameState] = useState("");
+
+  const [emailTimeNetworkingState, setEmailTimeNetworkingState] = useState(
+    Constants.FINISHED_MUTATION_TEXT,
+  );
+  // For email updating based on timeout
+  const updateEmailTimeout = useRef<number | undefined>();
+  const updateEmailHandler = () => {
+    updateEmailAlerts.mutate(hasEmailAlerts);
+  };
+
+  // update time when time is updated
+  const [isFirstRender, setIsFirstRender] = useState(false);
+  useEffect(() => {
+    if (isFirstRender) {
+      setIsFirstRender(false);
+    } else {
+      clearTimeout(updateEmailTimeout.current);
+      updateEmailTimeout.current = setTimeout(updateEmailHandler, 500);
+    }
+  }, [emailTime]);
+
+  // DANGEROUS delete account mutation
+  const deleteAccountMutation = useMutation({
+    mutationFn: () => api.delete("/user/"),
+    onSuccess: () => {
+      deleteUser();
+    },
+  });
+
+  // Updating email alerts setting
+  const updateEmailAlerts = useMutation({
+    // Only way to have optional parameters for mutationFn
+    // https://github.com/TanStack/query/issues/4264#issuecomment-1268054812
+    mutationFn: (enabled: boolean) =>
+      api
+        .post<SettingShort>("/user/settings/", {
+          key: "notify",
+          value: emailTime,
+          enabled: enabled,
+        })
+        .then((res) => res.data),
+    onSuccess: (data) => {
+      setHasEmailAlerts(data.enabled);
+      setEmailTime(data.value);
+      setEmailTimeNetworkingState(Constants.FINISHED_MUTATION_TEXT);
+    },
+  });
+
+  const { isLoading: usernameChangeLoading, mutate: changeUsername } =
+    useMutation({
+      mutationFn: (username: string) =>
+        api.put("/user/info/", {
+          data_type: "username",
+          data: username,
+        }),
+      onError: () => {
+        setChangeUsernameState("Username already taken.");
+      },
+      onSuccess: () => {
+        setChangeUsernameState("Username updated. Refresh to see changes.");
```

Comment:
maybe use a toast for this to make it more obvious?

Reply:
I'm following the design, I considered a toast but I feel like this is pretty clear imo. We can ask design on Tuesday as well.